### PR TITLE
Add pmreorder support

### PR DESCRIPTION
**Pull request by igchor**
and create on test for pmem::obj::array using pmreorder

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/110)

### CODE REVIEW

File: tests/helpers.cmake [hunk 1]
```diff
@@ -190,40 +190,109 @@ function(execute_common output_file name)
...
+# Optional function arguments are passed as consecutive arguments to
+# the command.
+function(pmreorder_execute expect_success engine conf_file name)
+    check_target(${name})
+
+    if(NOT (${TRACER} STREQUAL none))
+        message(FATAL_ERROR "Pmreorder test must be run without any tracer.")
+    endif()
+
+    set(ENV{PMEMOBJ_COW} 1)
```
Comment (pbalcer) [inline tests/helpers.cmake]: Is this required for pmreoder? I don't think so.

File: tests/ctest_helpers.cmake [hunk 2]
```diff
@@ -92,6 +92,37 @@ endif()
...
+				OUTPUT_QUIET)
+
+		string(REGEX MATCH ".*pmemcheck-([0-9.]*),.*" PMEMCHECK_OUT "${PMEMCHECK_OUT}")
+		set(PMEMCHECK_VERSION ${CMAKE_MATCH_1} CACHE INTERNAL "")
+	endif()
+endfunction()
+
+function(find_pmreorder)
+	set(ENV{PATH} ${LIBPMEMOBJ_PREFIX}/bin:$ENV{PATH})
+	execute_process(COMMAND pmreorder --help
```
Comment (marcinslusarz) [inline tests/ctest_helpers.cmake]: find_program?

File: utils/docker/images/install-pmdk.sh [hunk 3]
```diff
@@ -38,9 +38,9 @@ set -e
 
 git clone https://github.com/pmem/pmdk
 cd pmdk
-git checkout 9b8a2bec65c9eb4922841d53d696463a503842ee
+git checkout stable-1.5
```
Comment (marcinslusarz) [inline utils/docker/images/install-pmdk.sh]: "stable-1.5" is a moving target, please use the latest commit id from this branch

**Comment by marcinslusarz** [comment]
I think we need to add proper (always compiled-in) COW support on pmemobj side. I'm not sure it should block this PR though...

File: utils/docker/images/install-pmdk.sh (hunk 3 above)
Comment (igchor) [reply utils/docker/images/install-pmdk.sh]: Done.

File: tests/ctest_helpers.cmake (hunk 2 above)
Comment (igchor) [reply tests/ctest_helpers.cmake]: Done.

**Pull request merged**